ec2_metadata.py

```diff
@@ -61,6 +61,28 @@ def mac(self):
         return requests.get(METADATA_URL + 'mac').text
 
     @cached_property
+    def network(self):
```

Comment:
how about `network_interfaces` as the name? Also it would be nicer to return a class of objects rather than dicts - `ec2_metadata.network_interfaces[0].mac_address` is cleaner and more discoverable